plugin.video.vrt.nu/resources/lib/vrtplayer/urltostreamservice.py

```diff
@@ -17,25 +15,96 @@ def __init__(self, vrt_base, vrtnu_base_url, kodi_wrapper):
+            """ Generates a propery license key value
+
+            # A{SSM} -> not implemented
+            # R{SSM} -> raw format
+            # B{SSM} -> base64 format
+            # D{SSM} -> decimal format
+
+            The generic format for a LicenseKey is:
+            |<url>|<headers>|<key with placeholders|
+
+            The Widevine Decryption Key Identifier (KID) can be inserted via the placeholder {KID}
+
+            @type keyUrl: str
+            @param keyUrl: the URL where the license key can be obtained
+
+            @type keyType: str
+            @param keyType: the key type (A, R, B or D)
+
+            @type keyHeaders: dict
+            @param keyHeaders: A dictionary that contains the HTTP headers to pass
+
+            @type keyValue: str
+            @param keyValue: i
+            @return:
+            """
+
+            header = ''
+            if keyHeaders:
+                for k, v in list(keyHeaders.items()):
+                    header = '{0}&{1}={2}'.format(header, k, requests.utils.quote(v))
+
+            if keyType in ('A', 'R', 'B'):
+                keyValue = '{0}{{SSM}}'.format(keyType)
+            elif keyType == 'D':
+                if 'D{SSM}' not in keyValue:
+                    raise ValueError('Missing D{SSM} placeholder')
+                keyValue = requests.utils.quote(keyValue)
+
+            return '{0}|{1}|{2}|'.format(keyUrl, header.strip('&'), keyValue)
+
     def get_stream_from_url(self, url):
-        (session, token) = super(UrlToStreamService, self)._get_session_and_token_from_()
+        if 'vrtnu/kanalen' in url:
+            token = super(UrlToStreamService, self)._get_token_from_()
+        else:
+            token = super(UrlToStreamService, self)._get_sessiontoken_from_()
+        xbmc.log(str(token), xbmc.LOGNOTICE)
         if token is not None:
             url = urlparse.urljoin(self._vrt_base, url)
-            url_response = session.get(url);
-            strainer = SoupStrainer('div', {'class': 'vrtvideo videoplayer'})
-            soup = BeautifulSoup(url_response.content, 'html.parser', parse_only=strainer)
-            vrt_video = soup.find(lambda tag: tag.name == 'div' and tag.get('class') == ['vrtvideo'])
-            
-            data_publication_id = vrt_video['data-publicationid']
-            data_video_id = vrt_video['data-videoid']
-            final_url = self._STREAM_URL_PATH.format(data_publication_id, data_video_id, token)
-
-            stream_response = session.get(final_url)
-            hls = self.__get_hls(stream_response.json()['targetUrls'])
-            subtitle = None
-            #if self._kodi_wrapper.get_setting('showsubtitles') == 'true':
-            #    subtitle = self.__get_subtitle(stream_response.json()['subtitleUrls'])
-            return helperobjects.StreamURLS(hls, subtitle)
+            htmlpage = requests.get(url).text
+            divregex = re.compile(r'<div class=\"vrtvideo\"(.*)\">')
```

Comment:
Lets not use Regexes to get specific elements => use beautifulsoup instead. It will be easier to read & maintain. Please adjust everywhere where you used a regex for scraping 👍